containers/daap-resync-unprocessed-files/src/var/task/resync_unprocessed_files.py

```diff
@@ -0,0 +1,99 @@
+import os
+
+import boto3
+from data_platform_logging import DataPlatformLogger
+
+s3 = boto3.client("s3")
+logger = DataPlatformLogger(
+    extra={
+        "image_version": os.getenv("VERSION", "unknown"),
+        "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+    }
+)
+s3_security_opts = {
+    "ACL": "bucket-owner-full-control",
+    "ServerSideEncryption": "AES256",
+}
+raw_data_bucket = os.environ.get("RAW_DATA_BUCKET", "")
+curated_data_bucket = os.environ.get("CURATED_DATA_BUCKET", "")
+log_bucket = os.environ.get("RAW_DATA_BUCKET", "")
+athena_load_lambda = os.environ.get("ATHENA_LOAD_LAMBDA", "")
+s3 = boto3.client("s3")
+
+
+def handler(event, context):
+    data_product_to_recreate = event.get("data_product", "")
+    raw_prefix = f"raw_data/{data_product_to_recreate}"
+    curated_prefix = f"curated_data/database_name={data_product_to_recreate}"
+    logger.info(f"Raw prefix: {raw_prefix}")
+    logger.info(f"Curated prefix: {curated_prefix}")
+    logger.write_log_dict_to_s3_json(bucket=log_bucket, **s3_security_opts)
+
+    # Check data product has associated raw data
+    raw_pages = get_data_product_pages(
+        bucket=raw_data_bucket,
+        data_product_prefix=raw_prefix,
+    )
+
+    # Find extraction timestamps in the raw area, and not in the curated area
+    paginator = s3.get_paginator("list_objects_v2")
+    curated_pages = paginator.paginate(
+        Bucket=curated_data_bucket, Prefix=curated_prefix
+    )
+
+    # key = "raw_data/data_product/table/extraction_timestamp=timestamp/file.csv"
+    raw_table_timestamps = set(
+        "/".join(item["Key"].split("/")[1:4])
+        for item in raw_pages.search("Contents")
+        if item["Size"] > 0
+    )
+
+    curated_table_timestamps = set()
+    for item in curated_pages.search("Contents"):
+        # key = "curated_data/database_name=data_product/table_name=table"
+        #       + "/extraction_timestamp=timestamp/file.parquet"
```

Comment:
You are making an assumption here that path will be in the fomat "curated_data/database_name=data_product/table_name=table"
+ "/extraction_timestamp=timestamp/file.parquet".  If the table structure is changed t will error. Can you do pattern matching to check if its correct format  or extract in some other way to get **database_name and table_name**